### DISCUSSION TITLE
Cannot dump Game Gear

### DISCUSSION THREAD

**Discussion by OpenAMOC** [sanni/cartreader#950] [Need help]
Asking for help with dumping Game Gear carts. I have built the adapter using the save the hero pcb and gameslot GG, tried to use it in the mega drive slot of my v5 OSCR using putty as a serial monitor. I select SMS/GG then GG retrode then read rom but the CRC32 identified is not correct and not found, as pictured below, and the resulting .gg dump will not play in the emulator fusion. I am a bit lost as to why the dumps are failing, I've tried 5 different carts to no avail. Any advice much appreciated!

![image](https://github.com/sanni/cartreader/assets/95543179/ccccd9b8-4c80-4f89-a2f6-6bb680dc79a5)

**Comment by Ancyker** [comment]
STH uses the other one, Retron I think.

**Reply by Ancyker** [reply]
@SAVE-THE-HERO

**Reply by OpenAMOC** [reply]
> STH uses the other one, Retron I think.

Thanks Ancyker, but I seem to have the same issue with the retron option. Please see pictured below for 512kb NFL 95 and 32kb Columns. 
![image](https://github.com/sanni/cartreader/assets/95543179/750e6d50-f5ee-4ffc-b983-f8233800acea)
![image](https://github.com/sanni/cartreader/assets/95543179/4d078589-239b-4454-b2eb-0b44f0b9c456)

I have now checked continuity between the contact pads on the STH pcb and the labelled contact points at the bottom of the Gameslot GG board, initially all seems to be correct. I ensured the gg.txt file was present on the SD card, and reflashed the arduino so find myself out of ideas.

**Reply by Ancyker** [reply]
> I have now checked continuity between the contact pads on the STH pcb and the labelled contact points at the bottom of the Gameslot GG board, initially all seems to be correct. I ensured the gg.txt file was present on the SD card, and reflashed the arduino so find myself out of ideas.

Is the CRC32 the same every time?

**Reply by OpenAMOC** [reply]
> > I have now checked continuity between the contact pads on the STH pcb and the labelled contact points at the bottom of the Gameslot GG board, initially all seems to be correct. I ensured the gg.txt file was present on the SD card, and reflashed the arduino so find myself out of ideas.
> 
> Is the CRC32 the same every time?

Yes, when using the retrode option, the CRC32 is the same for a given game each time. For example, NFL 95 is 081EB154 and World Series Baseball 95 is 90AC22F2. As you might imagine, when searching gg.txt for these values, nothing is found.

**Reply by Ancyker** [reply]
> Yes, when using the retrode option, the CRC32 is the same for a given game each time. For example, NFL 95 is 081EB154 and World Series Baseball 95 is 90AC22F2. As you might imagine, when searching gg.txt for these values, nothing is found.

I checked and it is supposed to use Retrode. What other games do you have? Do you Sonic, G-Loc, or Outrun Europa? Are you in the Discord?

**Reply by SAVE-THE-HERO** [reply]
Good morning sir.
Retrosix GG slots are not of good quality.
When we assemble them at our factory and inspect them, we find defective products at a rate of about 30 to 40%.
The main reason is that the slot itself is distorted and the terminal parts are separated.
We only send out all of our products that have been successfully Dump tested.
If you bought from my store we can resend new one to you.
Please contact us at the following e-mail address
savetheherobuilders@gmail.com

**Reply by OpenAMOC** [reply]
> Good morning sir. Retrosix GG slots are not of good quality. When we assemble them at our factory and inspect them, we find defective products at a rate of about 30 to 40%. The main reason is that the slot itself is distorted and the terminal parts are separated. We only send out all of our products that have been successfully Dump tested. If you bought from my store we can resend new one to you. Please contact us at the following e-mail address [savetheherobuilders@gmail.com](mailto:savetheherobuilders@gmail.com)

Good morning, that is a real shame but I'm very thankful for your explanation. Unfortunately I built it myself instead of buying from your store, which may well be my next step! Thanks for all you've done for this community 👍

**Reply by OpenAMOC** [reply]
> > Yes, when using the retrode option, the CRC32 is the same for a given game each time. For example, NFL 95 is 081EB154 and World Series Baseball 95 is 90AC22F2. As you might imagine, when searching gg.txt for these values, nothing is found.
> 
> I checked and it is supposed to use Retrode. What other games do you have? Do you Sonic, G-Loc, or Outrun Europa? Are you in the Discord?

I do have G-Loc and plan to join the discord, but at this rate I'm likely to buy a new adapter from STH instead. Many many thanks for your time! 😄

**Reply by SAVE-THE-HERO** [reply]
I think you can Dump it if you correct it so that all the contacts in the slots are connected.
Might be possible if the board can be fixed flat.
https://x.com/peca_port0/status/1708412011431002389

Or Joey GG cart flasher seems reliable.
https://bennvenn.myshopify.com/products/joeygg-cart-flasher
Good luck.

**Comment by PsyK0p4T** [comment]
If it reads the TMR SEGA header, then I assume the mode you selected is menus is the correct one. What you can check : 
- confirm you're using 5V mode
- clean the cart pins very well (you can open the cartridge to do it better)
- compare your dump with an internet one to see where the differences are with an hex editor

A few games can be tricky to dump, but G-Loc should be fine, it was tested already.

**Reply by Ancyker** [reply]
Yeah, I asked about G-Loc because I have that one.

**Reply by OpenAMOC** [reply]
Thank you both for your advice, I have just tried G-Loc, ensuring I'm using 5v and the cart is clean (even tried again after using contact cleaner), but no luck. 
![image](https://github.com/sanni/cartreader/assets/95543179/fe122a2f-e651-48d1-9776-80f5fc5f2afd)

**Reply by Ancyker** [reply]
> Thank you both for your advice, I have just tried G-Loc, ensuring I'm using 5v and the cart is clean (even tried again after using contact cleaner), but no luck.

Can you join Discord?

**Reply by OpenAMOC** [reply]
Sure, just joined

**Comment by Ancyker** [comment]
We resolved this in Discord. OP's adapter is now working :)

It was determined that pin GG:D7/OSCR:D29 was shorted to ground on the adapter due to a damaged solder mask on the back which caused a stuck bit. Clearing the short fixed the issue. Images below for reference in case anyone finds this post in the future.

Stuck bit:
![Comparison of hex dumps between a working OSCR and the one being troubleshooted](https://github.com/sanni/cartreader/assets/7105333/f28cd827-ee53-4a16-9779-76729fd6bffa)

Picture of the short to ground:
![A picture showing a pin on the back of the adapter shorted to ground due to a damaged solder mask](https://github.com/sanni/cartreader/assets/7105333/731c7f8f-d1ab-4b44-857b-a177390079c2)

**Reply by PsyK0p4T** [reply]
Nice find ! Glad this was solved :)
